internal/postrs/proof.go

```diff
@@ -30,9 +30,12 @@ func translateScryptParams(params config.ScryptParams) C.ScryptParams {
 	}
 }
 
-func GenerateProof(datadir string, challenge []byte, cfg config.Config, nonces uint, threads uint, powScrypt config.ScryptParams) (*shared.Proof, error) {
-	datadirPtr := C.CString(datadir)
-	defer C.free(unsafe.Pointer(datadirPtr))
+func GenerateProof(dataDir string, challenge []byte, cfg config.Config, nonces uint, threads uint, powScrypt config.ScryptParams) (*shared.Proof, error) {
+	// TODO(mafa): disabled for now (calling it more than once crashes the program)
+	// C.configure_logging(C.Trace)
```

Comment:
The idea was that `C.configure_logging()` should be called only once, on "start"

Reply:
I assumed so, I want to avoid making `GenerateProof`, `VerifyProof` and `Initialize` stateful, so I like the callback approach better 🙂